Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -1338,23 +1337,24 @@ void RetryEventBind (BreakEventInfo binfo)
 		/// </remarks>
 		internal protected void UnbindSourceFileBreakpoints (string fullFilePath)
 		{
+			Dictionary<BreakEvent, BreakEventInfo> breakpointsCopy;
 			var toUpdate = new List<BreakEventInfo> ();
 
 			lock (breakpoints) {
 				// Make a copy of the breakpoints table since it can be modified while iterating
-				var breakpointsCopy = new Dictionary<BreakEvent, BreakEventInfo> (breakpoints);
+				breakpointsCopy = new Dictionary<BreakEvent, BreakEventInfo> (breakpoints);
+			}
 
-				foreach (var bps in breakpointsCopy) {
-					if (bps.Key is Breakpoint bp && bps.Value.Status == BreakEventStatus.Bound) {
-						if (System.IO.Path.GetFullPath (bp.FileName) == fullFilePath)
-							toUpdate.Add (bps.Value);
-					}
+			foreach (var bps in breakpointsCopy) {
+				if (bps.Key is Breakpoint bp && bps.Value.Status == BreakEventStatus.Bound) {
+					if (System.IO.Path.GetFullPath (bp.FileName) == fullFilePath)
+						toUpdate.Add (bps.Value);
 				}
+			}
 
-				foreach (var be in toUpdate) {
-					breakpoints.Remove (be.BreakEvent);
-					NotifyBreakEventStatusChanged (be.BreakEvent);
-				}
+			foreach (var be in toUpdate) {
+				breakpoints.Remove (be.BreakEvent);
```

Comment:
I think we need a lock around this breakpoints.Remove.

Reply:
Eek, yea, I missed that...